tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ServiceWrapperCommand.java

```diff
@@ -69,48 +71,90 @@ private File checkStaticWrapperFile() throws IOException
             {
                 if( lib.exists() && lib.getName().endsWith( "portal-kernel.jar" ) )
                 {
-                    targetJarFile.add( lib );
+                    portalkernelJar = lib;
+                    break;
                 }
             }
 
             libFiles = FileListing.getFileListing( new File( bundleServerPath.append( "../osgi" ).toOSString() ) );
+            libFiles.add( portalkernelJar );
 
-            for( File lib : libFiles )
+            if( !libFiles.isEmpty() )
             {
-                if( lib.exists() && lib.getName().endsWith( "api.jar" ) )
+                for( File lib : libFiles )
                 {
-                    targetJarFile.add( lib );
+                    if( lib.getName().endsWith( ".lpkg" ) )
+                    {
+                        try(JarFile jar = new JarFile( lib ))
+                        {
+                            Enumeration<JarEntry> enu = jar.entries();
+
+                            while( enu.hasMoreElements() )
+                            {
+                                JarEntry entry = enu.nextElement();
+
+                                String name = entry.getName();
+
+                                if( name.contains( ".api" ) )
```

Comment:
The name of bundles located in *.lpkg such as `com.liferay.blog.api-3.0.1.jar`. I think use `contains(".api")` is more appropriate.